internal/manager/utils/kongconfig/root.go

```diff
@@ -91,10 +92,10 @@ func DBModeFromRoot(r Root) (string, error) {
 		return "", fmt.Errorf("invalid %q type, expected a string, got %T", dbModeKey, dbMode)
 	}
 
-	return dbModeStr, nil
+	return dataplaneutil.DBMode(dbModeStr), nil
```

Comment:
Please consider using a constructor that will validate the data and return a strongly typed enum.

Reply:
Added a constructor with validation.